erdgo/address.go

```diff
@@ -0,0 +1,84 @@
+package erdgo
+
+import (
+	"encoding/hex"
+	"fmt"
+	"math"
+	"regexp"
+
+	"github.com/btcsuite/btcutil/bech32"
+)
+
+const (
+	bech32Hrp        = "erd"
+	bech32AddressLen = 62
+	pubkeyLen        = 32
+)
+
+// IsValidBech32Address returns true if the provided parameter is a valid Elrond bech32 address and false otherwise
+func IsValidBech32Address(address string) bool {
+	hrp, bytes, err := bech32.Decode(address)
+	if err != nil || hrp != bech32Hrp {
+		return false
+	}
+	pubkey, err := bech32.ConvertBits(bytes, 5, 8, false)
+	if err != nil {
+		return false
+	}
+
+	return len(address) == bech32AddressLen && len(pubkey) == pubkeyLen
+}
+
+// IsValidPubkey returns true if the provided parameter is a valid pubkey and false otherwise
+func IsValidPubkey(strPubkey string) bool {
+	hexDigitsPerByte := 2
```

Comment:
As an alternative, could decode from hex, check for error and for length.

Reply:
Done